pkgs/build-support/build-fhs-userenv/chrootenv/chrootenv.c

```diff
@@ -0,0 +1,150 @@
+#define _GNU_SOURCE
+
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include <errno.h>
+#include <sched.h>
+#include <unistd.h>
+
+#define g_perr(s, err) g_error("%s#L%d: %s: %s", __func__, __LINE__, s, g_strerror(err))
+#define g_perrno(s) g_perr(s, errno)
```

Comment:
`g_` is a glib namespace, you should not define names starting with `g_`.

Reply:
Sure, and names themselves are also bad and not descriptive. I've pushed a new version with `fail` and `fail_if` macros: I think it separates concerns much better than previously used error macros, and names are much more straightforward.